src/libs/sfdk/dockervirtualmachine_p.h

```diff
@@ -0,0 +1,109 @@
+/****************************************************************************
+**
+** Copyright (C) 2019 Open Mobile Platform LLC.
+** Contact: http://jolla.com/
+**
+** This file is part of Qt Creator.
+**
+** GNU Lesser General Public License Usage
+** Alternatively, this file may be used under the terms of the GNU Lesser
+** General Public License version 2.1 as published by the Free Software
+** Foundation and appearing in the file LICENSE.LGPL included in the
+** packaging of this file.  Please review the following information to
+** ensure the GNU Lesser General Public License version 2.1 requirements
+** will be met: http://www.gnu.org/licenses/old-licenses/lgpl-2.1.html.
+**
+** Other Usage
+**
+** Alternatively, this file may be used in accordance with the terms and
+** conditions contained in a signed written agreement between you and
+** The Qt Company.
+**
+****************************************************************************/
+
+#pragma once
+
+#include "virtualmachine_p.h"
+
+namespace Sfdk {
+
+class DockerVirtualMachinePrivate;
+class SFDK_EXPORT DockerVirtualMachine : public VirtualMachine
+{
+    Q_OBJECT
+
+public:
+    explicit DockerVirtualMachine(const QString &name, QObject *parent = nullptr);
+    ~DockerVirtualMachine() override;
+
+    static QString staticType();
+    static QString staticDisplayType();
+    static void fetchRegisteredVirtualMachines(const QObject *context,
+            const Functor<const QStringList &, bool> &functor);
+    static bool isAvailable();
```

Comment:
I would prefer moving `isAvailable` up - it's the first and possibly last function to call after all :) Similarly for vbox.

Reply:
fixed